### PR TITLE
Add a context menu to plots in the workbench

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

The new context menu will allow quick access to change certain properties of the plot. Only one option exist so far - the option to change the axes type. 

**To test:**

* build and start the workbench
* create a line plot
* right click and try the axes options.
* enable fitting and check the fit options still work and the axes options also work too

Refs #24483 

*This does not require release notes* because **it is part of the unreleased workbench.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
OSX failure is a compiler warning that has made it to master.

File: qt/applications/workbench/workbench/plotting/figureinteraction.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (TTitcombe) [inline qt/applications/workbench/workbench/plotting/figureinteraction.py]: 2019

File: qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory UKRI,
```
Comment (TTitcombe) [inline qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py]: 2019

**Review by TTitcombe — CHANGES_REQUESTED** [review]
A couple of issues picked up during testing:
1. If you set the axes to either of the log y options, go onto fit and then back off fit, switch to either of the lin y options. The y axis is not scaled correctly
2. I plotted `GEM38370_Focussed_Legacy`. In log x, for spectrum 2 I get ticks for 0, 1, and 10, but 1-10 are further apart than 0-1. For other spectrum numbers I only get ticks for 0 and 1, therefore most of the x axis has no ticks, which makes it difficult/impossible to tell values

File: qt/applications/workbench/workbench/plotting/figureinteraction.py (hunk 1 above)
Comment (martyngigg) [reply qt/applications/workbench/workbench/plotting/figureinteraction.py]: Good catch!

**Review by TTitcombe — APPROVED** [review]

**Comment by TTitcombe** [comment]
This adds nice functionality to workbench plotting. The plot scaling and axis ticks generated by right clicking on the plot are equivalent to when done through plot settings.

**Pull request merged**